packages/tools/src/tools/OverlayGridTool.ts

```diff
@@ -0,0 +1,384 @@
+}
+
+/**
+ * Get the largest normal axis of the a viewport first imageId
+ * @param viewport
+ * @returns
+ */
+function getViewportImageIdNormalAxis(viewport) {
+  const imageIds = viewport.getImageIds();
+  const { rowCosines, columnCosines } = metaData.get(
+    'imagePlaneModule',
+    imageIds[0]
+  );
+  const scanAxisNormal = vec3.cross(vec3.create(), rowCosines, columnCosines);
+  return getNormalLargestAxis(scanAxisNormal);
+}
+
+export interface OverlayGridAnnotation extends Annotation {
+  data: {
+    sourceViewport;
+  };
+}
+
+/**
+ * @public
+ */
+class OverlayGridTool extends AnnotationDisplayTool {
+  static toolName;
+
+  public touchDragCallback: any;
+  public mouseDragCallback: any;
+  _throttledCalculateCachedStats: any;
+  editData: {
+    renderingEngine: any;
+    annotations: Map<string, Annotation>;
+  } | null = {} as any;
+  isDrawing: boolean;
+  isHandleOutsideImage: boolean;
+
+  constructor(
+    toolProps: PublicToolProps = {},
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        sourceViewportIds: [],
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+  }
+
+  _init = (): void => {
+    const renderingEngines = getRenderingEngines();
+    const renderingEngine = renderingEngines[0];
+
+    // Todo: handle this case where it is too soon to get the rendering engine
+    if (!renderingEngine) {
+      return;
+    }
+
+    let viewports = renderingEngine.getViewports();
+    viewports = filterViewportsWithToolEnabled(viewports, this.getToolName());
+
+    // const sourceViewport = renderingEngine.getViewport(
+    //   this.configuration.sourceViewportId
+    // ) as Types.IVolumeViewport;
+
+    this.configuration.sourceViewportIds.forEach((viewportId) => {
+      const sourceViewport = renderingEngine.getViewport(viewportId);
+      if (sourceViewport || sourceViewport.getImageData()) {
+        const { element } = sourceViewport;
+        const { viewUp, viewPlaneNormal } = sourceViewport.getCamera();
+
+        const FrameOfReferenceUID = sourceViewport.getFrameOfReferenceUID();
+        if (!this.editData?.annotations) {
+          this.editData.annotations = new Map();
+        }
+
+        let annotation = this.editData.annotations[viewportId];
+
+        if (!annotation) {
+          const newAnnotation: OverlayGridAnnotation = {
+            highlighted: true,
+            invalidated: true,
+            metadata: {
+              toolName: this.getToolName(),
+              viewPlaneNormal: <Types.Point3>[...viewPlaneNormal],
+              viewUp: <Types.Point3>[...viewUp],
+              FrameOfReferenceUID,
+              referencedImageId: null,
+            },
+            data: {
+              sourceViewport,
+            },
+          };
+
+          addAnnotation(newAnnotation, element);
+          annotation = newAnnotation;
+        }
+
+        this.editData.annotations.set(viewportId, annotation);
```

Comment:
You don't need to use editData as state manager has it and you can `get` again in the renderAnnotation